redash-searcher-sync/src/app.rs

```diff
@@ -215,10 +217,12 @@ impl App {
         let mut oldest_updated_at = DateTime::<Local>::MAX_UTC;
         for query in res.results {
             let data_source = data_sources.get(&query.data_source_id).unwrap();
+            let tmp_query = query.query.clone();
             let doc = RedashDocument {
                 id: query.id,
                 name: query.name,
                 query: query.query,
+                references: parser::parse_references_from_query(tmp_query.as_str()),
```

Comment:
Does not this work? I think clone is not necessary for this case.

```suggestion
                references: parser::parse_references_from_query(&query.query),
```

Reply:
It cannot be done. Because the ownership of `query.query` has already moved in L224 `query: query.query,`. Hence, using a duplicate to get other ownership.